game-pathfinderwrathoftherighteous/index.js

```diff
@@ -1,46 +1,134 @@
-}
+const winapi = require('winapi-bindings');
+const {actions, fs, util} = require('vortex-api');
+
+const IsWin = process.platform === 'win32';
+
+const NexusId = 'pathfinderwrathoftherighteous';
+const Name = 'Pathfinder: Wrath\tof the Righteous';
+const ExeName = 'Wrath.exe';
+const SteamId = '1184370';
+const GogId = '1207187357';
+
+const ummDll = 'UnityModManager.dll';
+const ummModInfo = 'Info.json';
+
 function main(context) {
-    context.requireExtension('modtype-umm');
-    context.registerGame({
-        id: GAME_ID,
-        name: NAME,
-        logo: 'gameart.jpg',
-        mergeMods: true,
-        queryPath: findGame,
-        queryModPath: () => 'Mods',
-        executable: () => 'Wrath.exe',
-        requiredFiles: ['Wrath.exe'],
-        environment: {
-            SteamAPPId: STEAM_ID,
-        },
-        details: {
-            steamAppId: +STEAM_ID,
-        },
-        setup,
+  context.requireExtension('modtype-umm');
+  context.registerGame(
+    {
+      id: NexusId,
+      name: Name,
+      logo: 'gameart.jpg',
+      mergeMods: true,
+      queryPath: findGame,
+      queryModPath: () => 'Mods',
+      executable: () => ExeName,
+      requiredFiles: [ExeName],
+      environment: {
+        SteamAPPId: SteamId,
+      }, 
+      details:
+      {
+        steamAppId: SteamId,
+      },
+      setup: setup,
+    });
+  context.registerInstaller(NexusId + '-mod', 25, testMod, installMod);
+
+  function findGame() {
+    return util.steam.findByAppId(SteamId)
+      .then(game => game.gamePath)
+      .catch(() => readRegistryKey('HKEY_LOCAL_MACHINE',
+        `SOFTWARE\\WOW6432Node\\GOG.com\\Games\\${GogId}`,
+        'PATH'))
+      .catch(() => readRegistryKey('HKEY_LOCAL_MACHINE',
+        `SOFTWARE\\GOG.com\\Games\\${GogId}`,
+        'PATH'))
+  }
+
+  function readRegistryKey(hive, key, name) {
+    if (!IsWin) {
+      return Promise.reject(new util.UnsupportedOperatingSystem());
+    }
+
+    try {
+      const instPath = winapi.RegGetValue(hive, key, name);
+      if (!instPath) {
+        throw new Error('empty registry key');
+      }
+      return Promise.resolve(instPath.value);
+    } catch (err) {
+      return Promise.reject(new util.ProcessCanceled(err));
+    }
+  }
+
+  function findUnityModManager() {
+    return readRegistryKey('HKEY_CURRENT_USER', 'Software\\UnityModManager', 'Path')
+      .then(value => fs.statAsync(path.join(value, ummDll)));
+  }
+
+  function setup(discovery) {
+    return fs.ensureDirWritableAsync(path.join(discovery.path, 'Mods'), () => Promise.resolve())
+      .then(() => findUnityModManager()
+        .catch(err => {
+          return new Promise((resolve, reject) => {
+            context.api.store.dispatch(
+              actions.showDialog(
+                'question',
+                'Action required',
+                { message: 'You must install Unity Mod Manager to use mods with ' + Name + '.' },
```

Comment:
This dialog should not be necessary thanks to the modtype-umm extension which we are highlighting as a requirement